### PR TITLE
Bug: pointer -> to stack variable - fix with parent *name*

### PR DESCRIPTION
**Pull request by joakim-hove**
Now this was a really bad one - surprising it has not kicked before

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -2242,7 +2243,7 @@ void Schedule::invalidNamePattern( const std::string& namePattern,  std::size_t
         }
 
         for (const auto& gname : root_group.groups()) {
-            auto child_group = this->groupTree(gname, report_step, std::addressof(tree));
+            auto child_group = this->groupTree(gname, report_step, std::make_shared<GTNode>( tree ));
```
Comment (blattms) [inline src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:2246]: What your change is doing is creating a copy of the node for each entry in `root_group.groups()`. Seems like this means that trees in the `child_groups` point to different objects. Is that intended?

Wouldn't it better to change to `std::shared_ptr<GTNode> tree(root_group, parent)' in line 2237 and use that?

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:2246]: > Seems like this means that trees in the child_groups point to different objects.

Yes you are right.

> Is that intended?

No - that is not intended; it should not affect correctness - but it sure is a weird tree. 

> Wouldn't it better to change to `std::shared_ptr tree(root_group, parent)' in line 2237 and use that?

Maybe - I'll look into it; would like to not export the pointer semantics though.

**Comment by joakim-hove** [comment]
OK - this has been completely reworked; using the *name* of the parent we avoid all pointer semantics.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
Let me test whether this PR fixed my problem.

**Comment by GitPaean** [comment]
It fixes the problem reported before. Thanks.

**Pull request merged**